src/jitter.rs

```diff
@@ -785,5 +785,36 @@ impl RngCore for JitterRng {
 
 impl CryptoRng for JitterRng {}
 
-// There are no tests included because (1) this is an "external" RNG, so output
-// is not reproducible and (2) `test_timer` *will* fail on some platforms.
+#[cfg(test)]
+mod test_jitter_init {
+    use JitterRng;
+
+    #[cfg(feature="std")]
+    #[test]
+    fn test_jitter_init() {
+        use RngCore;
+        // Because this is a debug build, measurements here are not representive
+        // of the final release build.
+        // Don't fail this test if initializing `JitterRng` fails because of a
+        // bad timer (the timer from the standard library may not have enough
+        // accuracy on all platforms).
+        match JitterRng::new() {
+            Ok(ref mut rng) => {
+                // false positives are possible, but extremely unlikely
+                assert!(rng.next_u32() | rng.next_u32() != 0);
```

Comment:
Testing that two single `next_u32` samples are each not zero (to test the "half used" code) is probably about the only useful test to do here. Don't worry about 1 failure in 2^31 test runs; CI fails far *far* more often than that due to infrastructure / incompatible updates.

The other != 0 tests seem pointless to me.

Reply:
I mean `assert!(rng.next_u32() != 0)` twice to check *both* are not 0.